File: 2022/haskell/01/Main.hs
module Main (main) where

import Data.List (sortBy)
import Data.List.Split (splitOn)

groupCaloriesByElf :: [String] -> [[String]]
groupCaloriesByElf inputLines = splitOn [""] inputLines

readCaloriesAsInt :: [[String]] -> [[Int]]
readCaloriesAsInt = map (map read)

sumCaloriesByElf :: [[Int]] -> [Int]
sumCaloriesByElf caloriesByElf = map sum caloriesByElf

getTop3Elves :: [Int] -> [Int]
getTop3Elves sumCaloriesByElf = take 3 $ sortBy (flip compare) sumCaloriesByElf

main :: IO()
main = do
    inputFile <- readFile "../../inputs/01.txt"
    let top3Elves = getTop3Elves $ sumCaloriesByElf $ readCaloriesAsInt $ groupCaloriesByElf $ lines inputFile
    let result1 = top3Elves !! 0
    let result2 = sum top3Elves
    print result1
    print result2

File: 2022/python/01/main.py
elves = []

with open('../../inputs/01.txt','r') as input_file:
    elf = []
    for line in input_file:
        if line == '\n':
            elves.append(sum(elf))
            elf = []
        else:
            elf.append(int(line))

elves.sort(reverse=True)
result_1 = elves[0]
result_2 = sum(elves[:3])

print(result_1)
print(result_2)
